worker/peergrouper/machinetracker.go

```diff
@@ -105,7 +105,7 @@ func (m *machineTracker) SelectMongoAddressFromSpace(port int, space network.Spa
 	// If we end up here, then there are no addresses available in the
 	// specified space. This should not happen, because the configured
 	// space is used as a constraint when first enabling HA.
-	return "", fmt.Errorf("no addresses found for machine %q in space %q", m.id, space)
+	return "", nil
```

Comment:
Should we return an error, but with a special error type the calling code knows to look for? No error here feels wrong.

You can do something like:
type ErrNoAvailableAddresses struct {error}

And then have something like:

err := blah
if errors.Cause(err) == ErrNoAvailableAddresses {
}

I don't remember the exact syntax, because it isn't something we do often (we tend to end up hiding them behind a "IsNotFound()" sort of function).
But it does feel like a contract of just returning "" and no error is going to confuse people when they don't get what they wanted, and it wasn't considered an 'error'.

We could also return an error, but the caller just knows to batch them up and set statuses, instead of returning immediately.

Reply:
Yes; I'll change this.